### ISSUE TITLE
Android: Code doesn't always scroll horizontally

### ISSUE THREAD

**Issue by samuelclay** [samuelclay/NewsBlur#1076]
See https://newsblur.com/site/3667504/finding-the-missing-link

Scrolls on web and iOS but not Android.

<img width="739" alt="screen shot 2018-01-18 at jan 18 3 38 34 pm" src="https://user-images.githubusercontent.com/44229/35127169-a7730c24-fc65-11e7-8ada-d1bb46037be6.png">

![2018-01-18 15 37 12](https://user-images.githubusercontent.com/44229/35127195-c974d87a-fc65-11e7-8d4f-028b0b2519e2.png)

![2018-01-18 15 37 19](https://user-images.githubusercontent.com/44229/35127171-ac0611dc-fc65-11e7-9adc-53bc70fc501a.png)

**Comment by dosiecki** [comment]
The poor horizontal scrolling cababilities of the in-built browser component are generally incompatible with a left-to-right scrolling pager.  Horizontal scrolling is explicity disabled for this reason.  We have gone back and forth over the years on whether or not code should be wrapped.  The overall recommendation is that to view complex content such as code, the story should be opened in a real browser.  Short of implementing our own web browser or getting rid of left-right paging, there is not a whole lot we can do that won't break browsing on a tangible number of devices.

This has been brought up so many times that I am starting to wonder if the app needs an FAQ.

**Comment by dosiecki** [comment]
I should add that this is a great example of where Android's new open-in-app browser functionality can be used to good effect.  On moder devices, just tap the title of the story and it should open in a fully-functioning browser tab without even leaving the app where code should display just like in the full browser.  Getting back to the app is as quick at tapping the close tab button at the top or hitting the 'back' button.

**Comment by samuelclay** [comment]
What happens when you style pre/code with overflow-x: scroll? It breaks paging for the entire page or just where there's code?

**Comment by dosiecki** [comment]
We have horizontal scrolling totally disabled at the engine level.  The side effects of enabling it are myriad and ugly, due in large part to the fact that many devices' browser impls don't respect mad-width, width, or overflow properties, resulting in many stories (not just those with wide text blocks, but with any kind of large images or embeds) trying to scroll horizontally, blocking the left-right swipe action entirely, or causing bad screen flicker as the two scrolling portals fight to handle the drag events.

**Comment by dosiecki** [comment]
Just as a sanity check, I tried `overflow-x: scroll` with horizontal scrolling enabled on a few devices of various age and the results are still quite bad.  Essentially, the scrolling handler fights with the left/right paging handler and we get a few pixels of jittery horizontal scrolling followed by a glitchy and often incomplete attempt at swiping pages.  All of this is implemented down at the framework level, where we have reasonably little control.

One thing I was able to finally fix, however, is sorting out why our `word-wrap: break-word !important` property on `pre` blocks was being ignored.   It seems like many versions of the Android webview engine were defaulting some other properties oddly, so I'm patching them so `pre`'d content is at least 100% visible rather than clipped.
